weed/storage/volume_checking.go

```diff
@@ -12,28 +12,28 @@ func getActualSize(size uint32) int64 {
 	return NeedleHeaderSize + int64(size) + NeedleChecksumSize + int64(padding)
 }
 
-func CheckVolumeDataIntegrity(v *Volume, indexFile *os.File) (int64, error) {
+func CheckVolumeDataIntegrity(v *Volume, indexFile *os.File) (error) {
```

Comment:
I only removed the return value which was used to initials Volume.dataFileSize. I did not change the logic of the integrity check. The problem was that the returned value was 0 if the last needle was a tombstone, so the dataFileSize was then set to 0 which was definitively wrong. But because I removed the whole dataFileSize property the return value was then unused anyway and can be completely removed.